x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateParseTests.java

```diff
@@ -156,15 +157,64 @@ public void testInvalidPattern() {
                 new DateParse(
                     Source.EMPTY,
                     new Literal(Source.EMPTY, new BytesRef(pattern), DataType.KEYWORD),
-                    field("str", DataType.KEYWORD)
+                    field("str", DataType.KEYWORD),
+                    null
                 )
             ).get(driverContext)
         );
         assertThat(e.getMessage(), startsWith("invalid date pattern for []: Invalid format: [" + pattern + "]"));
     }
 
+    public void testInvalidLocale() {
```

Comment:
Can you add those tests as `TestCaseSupplier`s in the `parameters()` method? They automatically execute a bunch of cases.
I'm not sure of how map parameters work there really, but if it works like other params,you would be able to just test that it returns `null` with a warning with the error, which is what the user would see

Reply:
actually it's different because this is not just a per-row warning. It actually fails the query. Since they are static this isn't a recoverable error.